src/main/java/io/jenkins/plugins/orka/AgentTemplate.java

```diff
@@ -172,10 +178,14 @@ public OrkaProvisionedAgent provision() throws IOException, FormException {
             }
 
             String host = this.parent.getRealHost(response.getHost());
+            String vmId = response.getId();
 
-            return new OrkaProvisionedAgent(this.parent.getDisplayName(), response.getId(), response.getHost(), host,
+            String agentName = (this.prefixString.length() > 0) ? vmId : this.prefixString + '_' + vmId;
```

Comment:
You can use `StringUtils.isNotBlank` for the check. It will also handle the cases where `this.perfixString` is null in case it ever is.

Reply:
Btw you should invert the statement ;) You want to do exactly the opposite - use the prefix when defined and not used it when not defined :)